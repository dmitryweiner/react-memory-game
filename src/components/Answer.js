import React, {Component} from "react";
import Countdown360 from "react-countdown360";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import {Check} from "@material-ui/icons";

class Answer extends Component {
  state = {
    userAnswer: ""
  };

  handleInputChange = (event) => {
    this.setState({
      userAnswer: event.target.value
    });
  };

  handleKeyPress = (event) => {
    if (event.key === 'Enter') {
      this.handleAnswerComplete();
      event.preventDefault();
    }
  };

  handleAnswerComplete = () => {
    this.props.handleAnswerComplete(this.state.userAnswer);
  };

  render() {
    return (
      <>
        <Countdown360
          seconds={this.props.delay}
          onComplete={this.handleAnswerComplete}
        />
        <TextField
          data-testid="answer-input"
          autoFocus
          value={this.state.userAnswer}
          onChange={this.handleInputChange}
          onKeyPress={this.handleKeyPress}
          label="Введите запомненные числа"
          helperText="Введите в любой последовательности и нажмите Enter либо кнопку справа"
          variant="outlined"
        />
        <Button
          data-testid="answer-complete-button"
          variant="contained"
          color="primary"
          onClick={this.handleAnswerComplete}
        >
          <Check/>
        </Button>
      </>
    );
  }
}

export default Answer;
