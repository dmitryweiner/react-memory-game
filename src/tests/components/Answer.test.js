import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from "react-test-renderer";
import Answer from "../../components/Answer";


test('Answer calls proper handler when pressing button', async () => {
  const props = {
    delay: 10,
    handleAnswerComplete: jest.fn(),
  };
  const callback = props.handleAnswerComplete;
  const userInput = "1 2 3";

  const { getByTestId } = render(<Answer {...props}/>);
  const buttonElement = getByTestId("answer-complete-button");
  const inputElement = getByTestId("answer-input");
  expect(buttonElement).toBeInTheDocument();
  expect(inputElement).toBeInTheDocument();
  expect(callback).not.toBeCalled();

  fireEvent.change(
    inputElement.querySelector('input'),
    { target: { value: userInput } }
  );
  fireEvent.click(buttonElement);
  expect(callback).toBeCalled();
  expect(callback).toHaveBeenCalledTimes(1);
  expect(callback).toHaveBeenCalledWith(userInput);
});

test("Answer", () => {
  const props = {
    delay: 10,
    handleAnswerComplete: jest.fn(),
  };
  const component = renderer.create(<Answer {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});
