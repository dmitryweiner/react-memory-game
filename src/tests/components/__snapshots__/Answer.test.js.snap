// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer 1`] = `
Array [
  <div
    className="sc-fzplWN hsXBhy"
  >
    <div
      className="sc-fznyAO kaXgpD"
    >
      10
      <br />
      <div
        className="sc-fznKkj jDnUFy"
      >
        seconds
      </div>
    </div>
  </div>,
  <div
    className="MuiFormControl-root MuiTextField-root"
    data-testid="answer-input"
    onKeyPress={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
    >
      Введите запомненные числа
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={true}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-7"
        >
          <span>
            Введите запомненные числа
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-contained"
    >
      Введите в любой последовательности и нажмите Enter либо кнопку справа
    </p>
  </div>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-testid="answer-complete-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
]
`;
